slack-server-integration-common/src/main/java/com/atlassian/plugins/slack/settings/DefaultSlackSettingsService.java

```diff
@@ -29,11 +30,11 @@
     private final TransactionTemplate transactionTemplate;
 
     @Override
-    public boolean isChannelMuted(final String channelId) {
+    public boolean isChannelMuted(final ConversationKey conversationKey) {
         boolean isMuted = false;
-        if (StringUtils.isNotBlank(channelId)) {
+        if (StringUtils.isNotBlank(conversationKey.toString())) {
             List<String> mutedChannelIds = getMutedChannelIds();
-            isMuted = mutedChannelIds.contains(channelId);
+            isMuted = mutedChannelIds.contains(conversationKey);
```

Comment:
Here presence of `ConversationKey` value is checked in list of type `List<String>`. For some reasons `.contains()` method doesn't use generic type of the `List`, so compiler doesn't raise an error here during the compilation. Nevertheless, the result of this invocation will always be `false`. 

To address it I'd suggest to use `<teamId>:<channelId>` as a value for muting channels. It may be implemented as a new method `toStringKey()` in `ConversationKey` class. I'd like it to be addressed in this PR because it makes muting logic partially broken.

Reply:
That's right. Yet another change that will be necessary.